packages/vite/src/index.ts

```diff
@@ -142,6 +142,9 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
 
         if (cacheOptions.enabled && (statSync(`${cacheOptions.dir}/${id}`, { throwIfNoEntry: false })?.size ?? 0) > 0) {
           metadata = (await sharp(`${cacheOptions.dir}/${id}`).metadata()) as ImageMetadata
+          // fix avif metadata format https://github.com/lovell/sharp/issues/2504 https://github.com/lovell/sharp/issues/3746
```

Comment:
I think this doesn't explain what's happening very well. how about this:
```suggestion
          // we set the format on the metadata during transformation using the format directive
          // when restoring from the cache, we use sharp to read it from the image and that results in a different value for avif images
          // see https://github.com/lovell/sharp/issues/2504 and https://github.com/lovell/sharp/issues/3746
```